Code/GraphMol/SubstanceGroup.cpp

```diff
@@ -275,26 +275,29 @@ bool SubstanceGroup::includesBond(unsigned int bondIdx) const {
   return false;
 }
 
-bool SubstanceGroupChecks::isValidType(const std::string &type) {
-  return std::find(SubstanceGroupChecks::sGroupTypes.begin(),
-                   SubstanceGroupChecks::sGroupTypes.end(),
-                   type) != SubstanceGroupChecks::sGroupTypes.end();
+namespace SubstanceGroupChecks {
+const std::vector<const char *> sGroupTypes = {
```

Comment:
consider using `std::array<const char *, 15>`

Reply:
Given that it introduces a need to explicitly change the size if we ever add/remove supported types, what's the actual benefit of using an array instead of a const vector here?

Reply:
std::array doesn't have dynamic initialization so this cost is avoided.  Since we are copying and only performing std::find, I think this is a preferable data structure.

Reply:
It's the C++ equivalent of const char * array[15] = {....} with the benefit that the data structure knows the size of the array so most of the std functions work on it.

Reply:
Is the cost of dynamic initialization of a const global really something we should think about? That smells very, very strongly of premature optimization to me

Reply:
In this case, probably not since it's only allocated once and we are already pointing to const char * and copying the pointers is cheap.   However, IMO we aren't using the extra features of std::vector here so we shouldn't pay for them.

And I agree if this were in a function, I'd push harder.  In any case  I'm really ready for automatic choice of heap/stack dynamic arrays since these can really speed up functions.

Reply:
what do you mean here? Do you want the change made in order for this PR to be merged or not?

Reply:
@bp-kelley I need an answer here

Reply:
Sorry I missed this, “here” means we aren’t using any functionality that std::vector provides that isn’t supported by std::array

Reply:
Sorry, that wasn't the question: does the change need to be made for a merge?

Reply:
No, I'm fine with either.